httpcore5/src/main/java/org/apache/hc/core5/http/io/entity/EntityUtils.java

```diff
@@ -107,12 +117,8 @@ public static void consume(final HttpEntity entity) throws IOException {
             if (inStream == null) {
                 return null;
             }
-            int i = (int) Args.checkContentLength(entity);
-            if (i < 0) {
-                i = 4096;
-            }
-            final ByteArrayBuffer buffer = new ByteArrayBuffer(i);
-            final byte[] tmp = new byte[4096];
+            final ByteArrayBuffer buffer = new ByteArrayBuffer(getCheckedContentLength(entity));
```

Comment:
What if `Content-Length` is `500_000_000`? Will you really create a buffer with that size?

Reply:
@michael-o 
That's what the currently release code does; please see https://github.com/apache/httpcomponents-core/blob/rel/v4.4.12/httpcore/src/main/java/org/apache/http/util/EntityUtils.java#L135

May we address this separately please?

Reply:
Yes, we should. It requires at least warning/documentation.

Reply:
@michael-o Anyone in their mind should never be using `EntityUtils#toString` in the first place. I do not think there is anything to be done here other than saying that one ought not use this method in production.

Reply:
@ok2c That would be perfectly fine. Nothing can be improved here codewise.

Reply:
Would it then be appropriate to have a method that takes a max result size argument?

Reply:
As an overload, likely. How do you intend the fail if content length is greater or chunked transfer has reached that limit?

Reply:
@michael-o You get what you asked for, a String of length N. The call site can decide what to do if that length is less than the content length or some other length specifier.

Reply:
@garydgregory I propose to commit this change-set and raise another PR for the new method.

Reply:
+1